cpp/src/randomforest/randomforest.cuh

```diff
@@ -46,51 +46,28 @@ namespace ML {
 
 template <class T, class L>
 class RandomForest {
- private:
-  DT::DecisionTree<T, L>* trees = nullptr;  // pointer to trees of RF
-
-  /**
-   * @brief Return a const pointer to decision trees.
-   * @tparam T: data type for input data (float or double).
-   * @tparam L: data type for label data (int for classification task; float or double for
-   * regression task)
-   */
-  const DT::DecisionTree<T, L>* get_trees_ptr() const { return trees; }
-
  protected:
   RF_params rf_params;  // structure containing RF hyperparameters
   int rf_type;          // 0 for classification 1 for regression
 
-  /**
-   * @brief Sample row IDs for tree fitting and bootstrap if requested.
-   * @param[in] tree_id: unique tree ID
-   * @param[in] n_rows: total number of data samples.
-   * @param[in] n_sampled_rows: number of rows used for training
-   * @param[in, out] selected_rows: already allocated array w/ row IDs
-   * @param[in] num_sms: No of SM in current GPU
-   * @param[in] stream: Current cuda stream
-   * @param[in] device_allocator: Current device allocator from cuml handle
-   */
-  void prepare_fit_per_tree(int tree_id,
-                            int n_rows,
-                            int n_sampled_rows,
-                            unsigned int* selected_rows,
-                            const int num_sms,
-                            const cudaStream_t stream,
-                            const std::shared_ptr<raft::mr::device::allocator> device_allocator)
+  void get_row_sample(int tree_id,
+                      int n_rows,
+                      MLCommon::device_buffer<int>* selected_rows,
+                      const cudaStream_t stream,
+                      const std::shared_ptr<raft::mr::device::allocator> device_allocator)
   {
     ML::PUSH_RANGE("bootstrapping row IDs @randomforest.cuh");
-    int rs = tree_id;
-    if (rf_params.seed != 0) rs = rf_params.seed + tree_id;
 
-    raft::random::Rng rng(rs * 1000 | 0xFF00AA, raft::random::GeneratorType::GenKiss99);
+    // Hash these together so they are uncorrelated
+    int rs = DT::fnv1a32(rf_params.seed, tree_id);
```

Comment:
```suggestion
    int rs = DT::fnv1a32_basis;
    rs = fnv1a32(rs, rf_params.seed);
    rs = fnv1a32(rs, tree_id);
```
This would be correct way to use FNV hash.